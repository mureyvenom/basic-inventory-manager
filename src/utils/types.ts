import { ReactNode } from 'react';

export interface InventoryItem {
  name: string;
  stock: number;
  price: number;
  description: string;
  owner: string;
  date_created: string;
  last_modified: string;
}

export interface UserContextType {
  user: {
    email: string;
  };
  setUser: (e: string) => void;
}

export interface InventoryContextType {
  items: InventoryItem[] | null;
  saveItems: (items: InventoryItem[]) => void;
  addItem: (item: InventoryItem) => boolean;
  updateItem: (original: InventoryItem, replace: InventoryItem) => boolean;
  deleteItem: (item: InventoryItem) => void;
  getItems: () => InventoryItem[];
}

export interface SettingsContextType {
  darkmode: boolean;
  toggleDarkMode: () => void;
}

export interface ProviderProps {
  children: ReactNode;
}
