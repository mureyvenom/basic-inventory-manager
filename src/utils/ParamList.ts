import { InventoryItem } from './types';

export type MainStack = {
  Signin: undefined;
  InventoryList: undefined;
  Inventory: {
    mode: 'add' | 'edit';
    inventory?: InventoryItem;
  };
};
