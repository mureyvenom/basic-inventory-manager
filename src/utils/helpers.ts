export const numberWithCommas = (x: number, decimals?: number) => {
  return x.toFixed(decimals || 0).replace(/\B(?=(\d{3})+(?!\d))/g, ',');
};

export const addAlpha = (color: string, opacity: number) => {
  // coerce values so ti is between 0 and 1.
  var _opacity = Math.round(Math.min(Math.max(opacity || 1, 0), 1) * 255);
  return color + _opacity.toString(16).toUpperCase();
};

export const appConstants = () => {
  return {
    inventoryItemsKey: 'inventory-app-items',
    inventoryUserKey: 'inventory-app-user',
    inventorySettingsKey: 'inventory-app-settings',
  };
};

export const mockInventoryItems = [
  {
    date_created: '2023-03-11T02:27:30.776Z',
    description: 'Testing this out',
    last_modified: '2023-03-11T02:46:15.243Z',
    name: 'Testing this out',
    owner: 'holuwamurewa@gmail.com',
    price: 250,
    stock: 10,
  },
  {
    date_created: '2023-03-11T02:42:41.800Z',
    description: 'Second test',
    last_modified: '2023-03-11T02:48:05.826Z',
    name: 'Second test',
    owner: 'holuwamurewa@gmail.com',
    price: 1200,
    stock: 50,
  },
  {
    date_created: '2023-03-11T02:45:20.812Z',
    description: 'description',
    last_modified: '2023-03-11T02:45:20.812Z',
    name: 'Fourth entry',
    owner: 'holuwamurewa@gmail.com',
    price: 780,
    stock: 1000,
  },
];
