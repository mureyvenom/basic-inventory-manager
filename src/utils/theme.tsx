import { createTheme } from '@shopify/restyle';

const palette = {
  blue: '#6294CD',
  green: '#62CDA6',
  red: '#F92A4C',
  lightgreen: 'rgba(73,217,140,0.25)',
  lightred: 'rgba(255,102,127,0.25)',
  lightBg: 'rgba(255,255,255,0.05)',
  gray: '#2A3B3B',
  darkred: '#DA5757',
  darkgray: '#707070',
  black_: '#0A0A0A',
  lightBlue: '#29B9DC',
  purple: '#ED62F6',
  white: '#DDDDDD',
  muted: '#B0B0B0',
  faint: '#151515',
  transparent: 'transparent',
  constant_white: '#ffffff',
  constant_black: '#000000',
  // light theme colors
  bgLight: '#ffffff',
  faintLight: '#F9F9F9',
};

export const theme = createTheme({
  colors: {
    background: palette.black_,
    foreground: palette.white,
    primary: palette.blue,
    success: palette.green,
    danger: palette.red,
    lightsuccess: palette.lightgreen,
    lightdanger: palette.lightred,
    dangerdark: palette.darkred,
    secondary: palette.gray,
    info: palette.lightBlue,
    light: palette.purple,
    muted: palette.muted,
    muteddark: palette.darkgray,
    faint: palette.faint,
    white: palette.constant_white,
    black: palette.constant_black,
  },
  spacing: {
    xs: 4,
    s: 8,
    m: 16,
    l: 24,
    xl: 40,
    xxl: 80,
  },
  textVariants: {
    bold: {
      color: 'foreground',
      fontWeight: '700',
    },
    medium: {
      color: 'foreground',
      fontWeight: '600',
    },
    regular: {
      color: 'foreground',
    },
  },
  breakpoints: {},
});

export type Theme = typeof theme;

export const lightTheme: Theme = {
  ...theme,
  colors: {
    ...theme.colors,
    background: palette.bgLight,
    faint: palette.faintLight,
    foreground: palette.black_,
  },
};

export default theme;
