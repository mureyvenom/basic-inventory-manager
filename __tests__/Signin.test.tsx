import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
// import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { ThemeProvider } from '@shopify/restyle';
import theme from '../src/utils/theme';
// import { render, fireEvent, act } from '@testing-library/react-native';
// import { MainStack } from '../src/utils/ParamList';
// import RootProvider from '../src/state/root.provider';
import Signin from '../src/screens/SigninScreen';
import { SettingsContext } from '../src/state/providers/settings.provider';
import { UserContext } from '../src/state/providers/user.provider';
import { InventoryContext } from '../src/state/providers/inventory.provider';
import { ProviderProps } from '../src/utils/types';

jest.useFakeTimers();

// let navigation: NativeStackNavigationProp<MainStack, 'Signin'>;

it('renders correctly', async () => {
  const RootProvider = ({ children }: ProviderProps) => {
    const darkmode = true;
    const toggleDarkMode = jest.fn();
    const addItem = jest.fn();
    const deleteItem = jest.fn();
    const getItems = jest.fn();
    const saveItems = jest.fn();
    const updateItem = jest.fn();
    const setUser = jest.fn();

    return (
      <SettingsContext.Provider
        value={{
          darkmode,
          toggleDarkMode,
        }}>
        <UserContext.Provider
          value={{
            setUser,
            user: {
              email: '',
            },
          }}>
          <InventoryContext.Provider
            value={{
              addItem,
              deleteItem,
              getItems,
              saveItems,
              updateItem,
              items: [],
            }}>
            {children}
          </InventoryContext.Provider>
        </UserContext.Provider>
      </SettingsContext.Provider>
    );
  };
  const tree = renderer
    .create(
      <RootProvider>
        <ThemeProvider theme={theme}>
          <Signin />
        </ThemeProvider>
      </RootProvider>,
    )
    .toJSON();
  await renderer.act(async () => expect(tree).toMatchSnapshot());
});
