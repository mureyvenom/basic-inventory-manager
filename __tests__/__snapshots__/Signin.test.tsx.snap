// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    style={
      {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#DDDDDD",
                "fontSize": 60,
                "marginBottom": 40,
                "marginTop": 80,
                "textAlign": "center",
              },
            ]
          }
        >
          Sign In
        </Text>
        <View
          style={
            [
              {
                "paddingHorizontal": 24,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "marginBottom": 24,
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#0A0A0A",
                    "borderColor": "#B0B0B0",
                    "borderRadius": 6,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "height": 50,
                    "paddingHorizontal": 16,
                  },
                ]
              }
            >
              <TextInput
                autoCapitalize="none"
                autoCorrect={false}
                keyboardType="email-address"
                onChangeText={[Function]}
                placeholder="Enter your email"
                placeholderTextColor="#B0B0B0"
                style={
                  {
                    "color": "#DDDDDD",
                    "flex": 1,
                    "height": "100%",
                  }
                }
                value=""
              />
            </View>
          </View>
          <View
            style={
              [
                {
                  "marginBottom": 24,
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "backgroundColor": "#0A0A0A",
                    "borderColor": "#B0B0B0",
                    "borderRadius": 6,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "height": 50,
                    "paddingHorizontal": 16,
                  },
                ]
              }
            >
              <TextInput
                onChangeText={[Function]}
                placeholder="Enter your password"
                placeholderTextColor="#B0B0B0"
                secureTextEntry={true}
                style={
                  {
                    "color": "#DDDDDD",
                    "flex": 1,
                    "height": "100%",
                  }
                }
                value=""
              />
            </View>
          </View>
          <View
            style={
              [
                {
                  "marginBottom": 16,
                  "marginTop": 24,
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "width": "100%",
                  },
                ]
              }
            >
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "backgroundColor": "#6294CD",
                        "borderRadius": 6,
                        "flexDirection": "row",
                        "height": 50,
                        "justifyContent": "center",
                        "paddingHorizontal": 24,
                      },
                      {},
                    ]
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#ffffff",
                          "fontSize": 16,
                          "fontWeight": "600",
                        },
                      ]
                    }
                  >
                    Sign In
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
